docs/recipes/Making-An-Api-Call.md

```diff
@@ -4,15 +4,276 @@
+        }
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to fetch Books:', e);
+        setData({ Books, isLoading: false, fetchError: true });
+      }
+    };
+    fetchBooks();
+  }, [id]);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (fetchError) {
+    return <p>Error fetching Books</p>;
+  }
+
+  return (
+    <div>
+      <h1>Books</h1>
+      <ul>
+        {
+          books && books.map((book) => (
+            <li key={book.id}>Title: { book.title }</li>
+          ))
+        }
+      </ul>
+    </div>
+  );
+};
+```
+
+This approach works for modules, however it will not fully benefit from server side rendering and,
+if the url is called across multiple modules, will result in duplicated API calls.
+
+## Server Side data fetching
+
+One App will attempt to render your module on the server before sending the resulting HTML
+back to client. If a module requires any asynchronous tasks to render, such as data fetching,
+then these will need to be performed before the One App server renders the module. To do this you can use [`loadModuleData`](https://github.com/americanexpress/one-app/blob/main/docs/api/modules/Loading-Data.md#moduleholocronloadmoduledata).
+
+`loadModuleData` will be invoked before a module is rendered on the One App server. This happens when modules are loaded using [`ModuleRoute`](https://github.com/americanexpress/holocron/tree/main/packages/holocron-module-route) or [`composeModules`](https://github.com/americanexpress/holocron/blob/main/packages/holocron/docs/api/README.md#composemodules). `loadModuleData` will also be called on the client when the Holocron module mounts and receives props.
+
+> You can read more about how to use `ModuleRoute` in [Routing-And-Navigation](./Routing-And-Navigation.md) and `composeModules` in the [Module-Composition](./Module-Composition.md) recipes.
+
+Here is an example using `loadModuleData` to server side data fetch for the above example:
+
+```jsx
+const loadModuleData = async ({ store, fetchClient, ownProps }) => {
+  store.dispatch({ type: 'FETCH_API' });
+  try {
+    const response = await fetchClient('https://some-data-server.com/api/v1/books');
+    if (response.ok) {
+      const data = await response.json();
+      store.dispatch({ type: 'LOADED_API', data });
+    } else {
+      store.dispatch({ type: 'FAILED_API' });
+    }
+  } catch (e) {
+    store.dispatch({ type: 'FAILED_API' });
+  }
+};
+
+Books.holocron = {
+  // Runs on both Server and Browser
+  loadModuleData,
+  reducer,
+};
+```
+
+The modules reducer would handle those dispatched actions so the module would be able to retrieve the data from the Redux store.
+
+## Duplicate Calls Across modules
+
+When adding response data to the redux store it can be tempting to try to \access this directly in other modules. This is **not** a recommended approach as you should aim to have modules as independent as possible.
```

Comment:
```suggestion
When adding response data to the redux store it can be tempting to try to access this directly in other modules. This is **not** a recommended approach as you should aim to have modules as independent as possible.
```